system-addon/content-src/asrouter/components/ModalOverlay/ModalOverlay.jsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+export class ModalOverlay extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  componentWillMount() {
+    this.setState({active: true});
+  }
+
+  componentWillReceiveProps() {
+    this.setState({active: true});
+  }
+
+  onClick(e) {
+    this.setState({active: false});
+    document.body.classList.remove("modal-open");
+  }
+
+  render() {
+    const {active} = this.state;
+    if (active) {
+      // This lets the body know that the modal is open so that we can hide the scroll bar
+      document.body.classList.add("modal-open");
```

Comment:
I think what you want here is a `componentDidUpdate` function, in which you would check the active state and set `document.body.classList` to the appropriate thing